distributed/diagnostics/progress.py

```diff
@@ -140,9 +140,18 @@ def stop(self, exception=None, key=None):
 class MultiProgress(Progress):
     """Progress variant that keeps track of different groups of keys
 
-    See Progress for most details.  This only adds a function ``func=``
-    that splits keys.  This defaults to ``key_split`` which aligns with naming
-    conventions chosen in the dask project (tuples, hyphens, etc..)
+    See Progress for most details.
+
+    Parameters
+    ----------
+
+    func : Callable
+        Function that splits keys. This defaults to ``key_split`` which
+        aligns with naming conventions chosen in the dask project (tuples,
+        hyphens, etc..)
+
+    spans : bool, default: False
+        Use spans to group keys. If True, `func` can't be used.
```

Comment:
I thought briefly about the API again. We could also deprecate `func` and replace it with something like

`group_by: Callable, Literal["span"], Literal["prefix"]`

default would be `prefix` which then falls back to `key_split`. If `span` is selected, we'd do the spans thing you are introducing.
If the user provides a callable, we just use this instead.

This would not allow the full power of `scheduler_func` but would be a compromise.

Reply:
Why do we need both `prefix` and `Callable`, if `prefix` is basically `key_split: Callable`?

Reply:
This suggestion is now implemented.